src/transformers/generation/utils.py

```diff
@@ -888,6 +896,40 @@ def _reorder_cache(self, past_key_values, beam_idx):
             f" enable beam search for {self.__class__}"
         )
 
+    def _get_candidate_generator(
+        self,
+        generation_config: GenerationConfig,
+        input_ids: torch.LongTensor,
+        inputs_tensor: torch.Tensor,
+        assistant_model: "PreTrainedModel",
+        logits_processor: LogitsProcessorList,
+        model_kwargs: Dict,
+    ) -> CandidateGenerator:
+        """
+        Returns the candidate generator to be used in `assisted_generation`
+        """
+        # Check if assistant_model is a string
+        if isinstance(assistant_model, str):
```

Comment:
I would check whether e.g. `prompt_lookup_num_tokens` is set. It will work if we default it to `None`.

Not relying on strings to set modes would go more in line with how `generate` works at the moment :)